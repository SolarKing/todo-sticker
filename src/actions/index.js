// @flow
import type { TodoActions } from './todo';
import type { FilterActions } from './filter';
import type { SortActions } from './sort';

export type Actions =
    | TodoActions
    | FilterActions
    | SortActions;
