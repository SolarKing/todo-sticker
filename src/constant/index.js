// @flow

const DueDate = {
    today: 'today',
    all: 'all'
};


export const FilterEnum = {
    DueDate
};
