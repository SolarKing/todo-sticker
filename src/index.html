<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="UTF-8">
	<title>Todo Sticker</title>
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
</head>
<body>
	<div id="todoModal"></div>
	<div id="app"></div>
</body>
</html>
