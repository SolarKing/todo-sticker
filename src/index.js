// @flow

import React from 'react';
import ReactDOM from 'react-dom';
import TodosPage from './component/todosPage';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from './reducers';
import '../node_modules/antd/dist/antd.min.css';
import '../node_modules/antd/dist/antd.min.js';
import type {TodoState} from './states/index';
import NotifyServer from './class/notify/notifyServer';

// localStorage.removeItem('todos');
const initStorageData = localStorage.getItem('todos') || '{}';
const store = createStore(rootReducer, JSON.parse(initStorageData));
store.subscribe(() => {
    localStorage.setItem('todos', JSON.stringify(store.getState()));
    console.log(store.getState());
});

let todayTimestamp: number = new Date().getTime();

store.getState().todos.forEach((todo: TodoState) => {
    if (todo.complete) {
        return;
    }
    if (todo.dueDatetime <= todayTimestamp) {
        return;
    }
    NotifyServer.instance.addSchedule(todo.id || 'empty', todo.dueDatetime, todo.description);
});

const app = document.getElementById('app');

app && ReactDOM.render(
    <Provider store={store}>
        <TodosPage />
    </Provider>,
    app
);
