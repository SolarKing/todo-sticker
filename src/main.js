// @flow
import stickerHandler from './sticker/stickerHandler';
const { app, BrowserWindow } = require('electron');

let win;

declare var __dirname: any;
declare var process: any;

function createWindow() {
    win = new BrowserWindow({ 
        width: 1024, 
        height: 768,
        minWidth: 600,
        minHeight: 480,
    });
    win.loadURL(`file://${__dirname}/app.html`);

    if (process.env.NODE_ENV === 'development') {
        win.webContents.openDevTools();
    }

    win.on('closed', () => {
        win = null;
    });

}

stickerHandler.registerListener();

app.on('ready', () => {
    createWindow();
    if (/^win/i.test(process.platform)) {
        app.setAppUserModelId('com.gavin.todosticker');
    }
});


app.on('window-all-closed', () => {
    if (process.platform !== 'darwin') {
        app.quit();
    }
});

app.on('activate', () => {
    if (win === null) {
        createWindow();
    }
});