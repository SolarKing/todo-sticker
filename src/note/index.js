// @flow
import React from 'react';
import ReactDOM from 'react-dom';
import NoteDescription from '../component/noteDescription';
import './index.scss';

const app = document.getElementById('app');

app && ReactDOM.render(
    <NoteDescription />,
    app
);


