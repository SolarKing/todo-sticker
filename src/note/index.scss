body {
    width: 100vw;
    height: 100vh;
    margin: 0;
    background-color: #f7f7a3;
}