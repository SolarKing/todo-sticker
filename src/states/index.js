// @flow
import type {DueDateFilter} from '../constant/filter';

export type TodoState = {|
    id?: number,
    notificationDate?: number,
    notificationTime?: number,
    description: string
|}

export type FilterState = {|
    dueDate: DueDateFilter
|}

export type AppState = {|
    todos: TodoState[],
    filter: FilterState
|}
