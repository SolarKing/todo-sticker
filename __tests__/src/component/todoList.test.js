// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import TodoList from '../../../src/component/todoList';
import { type TodoState } from '../../../src/states';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('TodoList', () => {
    let wrapper: ShallowWrapper;

    let todoLists: Array<TodoState> = [
        {
            id: '1',
            dueDatetime: 10,
            description: 'Hello!',
            complete: false,
            createTime: 0,
        },
        {
            id: '2',
            dueDatetime: 123,
            description: 'hi!',
            complete: false,
            createTime: 0,
        },
        {
            id: '3',
            dueDatetime: 0,
            description: 'Hello there!',
            complete: true,
            createTime: 0,
        },
        {
            id: '4',
            dueDatetime: 15,
            description: 'Yo!',
            complete: false,
            createTime: 0,
        },
        {
            id: '5',
            dueDatetime: 10,
            description: 'Good morning!',
            complete: false,
            createTime: 0,
        },
    ];

    beforeAll(() => {
        jest.spyOn(Date.prototype, 'getTime').mockReturnValue(0);
        wrapper = shallow(
            <TodoList
                todos={todoLists}
            />
        );
    });

    it('should render correct with empty', () => {
        wrapper.setProps({
            todoLists: []
        });
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correct with todos', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should handle page changed', () => {
        wrapper.find('PaginationRow').simulate('change', 2);
        expect(wrapper.state().page).toEqual(2);
    });

    it('should generate showing todos when page changed', () => {
        wrapper.find('PaginationRow').simulate('change', 2);
        expect(wrapper.find('todoList__ListRow').prop('dataSource')).toEqual([
            todoLists[3],
            todoLists[4],
        ]);
    });

    it('should auto change to last page when todos moved', () => {
        expect(wrapper.state().page).toEqual(2);
        wrapper.setProps({
            todos: [
                todoLists[0],
                todoLists[1],
                todoLists[2],
            ]
        });
        expect(wrapper.state().page).toEqual(1);
    });

});