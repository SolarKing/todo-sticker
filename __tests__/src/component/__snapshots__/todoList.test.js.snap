// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render correct with empty 1`] = `
<todoList__TodosListGrid>
  <todoList__ListRow
    dataSource={
      Array [
        Object {
          "complete": false,
          "createTime": 0,
          "description": "Hello!",
          "dueDatetime": 10,
          "id": "1",
        },
        Object {
          "complete": false,
          "createTime": 0,
          "description": "hi!",
          "dueDatetime": 123,
          "id": "2",
        },
        Object {
          "complete": true,
          "createTime": 0,
          "description": "Hello there!",
          "dueDatetime": 0,
          "id": "3",
        },
      ]
    }
    renderItem={[Function]}
  />
  <PaginationRow
    defaultPageSize={3}
    onChange={[Function]}
    total={5}
  />
</todoList__TodosListGrid>
`;

exports[`TodoList should render correct with todos 1`] = `
<todoList__TodosListGrid>
  <todoList__ListRow
    dataSource={
      Array [
        Object {
          "complete": false,
          "createTime": 0,
          "description": "Hello!",
          "dueDatetime": 10,
          "id": "1",
        },
        Object {
          "complete": false,
          "createTime": 0,
          "description": "hi!",
          "dueDatetime": 123,
          "id": "2",
        },
        Object {
          "complete": true,
          "createTime": 0,
          "description": "Hello there!",
          "dueDatetime": 0,
          "id": "3",
        },
      ]
    }
    renderItem={[Function]}
  />
  <PaginationRow
    defaultPageSize={3}
    onChange={[Function]}
    total={5}
  />
</todoList__TodosListGrid>
`;
